src/lib/AnnotationControls.ts

```diff
@@ -9,31 +12,32 @@ export const CLASS_REGION_BUTTON = 'bp-AnnotationControls-regionBtn';
 export const CLASS_BUTTON_ACTIVE = 'is-active';
 export const CLASS_GROUP_HIDE = 'is-hidden';
 
-export type RegionHandler = ({ isRegionActive, event }: { isRegionActive: boolean; event: MouseEvent }) => void;
+export type AnnotationMode = 'draw' | 'highlight' | 'region' | null;
+export type RegionHandler = ({ activeControl, event }: { activeControl: AnnotationMode; event: MouseEvent }) => void;
 export type Options = {
     onRegionClick?: RegionHandler;
 };
 
 declare const __: (key: string) => string;
 
 export default class AnnotationControls {
-    /** @property {Controls} - Controls object */
+    private annotator: EventEmitter;
+
     private controls: Controls;
 
-    /** @property {HTMLElement} - Controls element */
     private controlsElement: HTMLElement;
 
-    /** @property {boolean} - Region comment mode active state */
-    private isRegionActive = false;
+    private currentActiveControl: AnnotationMode = null;
 
     /**
      * [constructor]
      */
-    constructor(controls: Controls) {
+    constructor(controls: Controls, annotator: EventEmitter) {
         if (!controls || !(controls instanceof Controls)) {
             throw Error('controls must be an instance of Controls');
         }
 
+        this.annotator = annotator;
```

Comment:
We should avoid passing around and storing the annotator instance when possible. If it gets destroyed and recreated independently, this will probably break. Can the BaseViewer manager the event handlers and invoke methods on the annotation controls?

Reply:
> We should avoid passing around and storing the annotator instance when possible. If it gets destroyed and recreated independently, this will probably break. Can the BaseViewer manage the event handlers and invoke methods on the annotation controls?

Sounds good